### ISSUE TITLE
tile vibration during flyTo

### ISSUE THREAD

**Issue by jfirebaugh** [Leaflet/Leaflet#3194]
There is some tile vibration during flyTo -- rapid vertical iteration of a pixel or two up and down. It's most apparent in the zoom in portion of the animation, and more obvious if you change the tile fade animation to be longer, so that you can see some overzoomed labels.

**Comment by mourner** [comment]
Think that's a rounding issue?

**Comment by adgoncal** [comment]
Do you think this is related to #3197?

**Comment by mourner** [comment]
@adgoncal no, doesn't seem related.

**Comment by briantjacobs** [comment]
This vibration is still a rough issue, especially when flying short distances with a longer duration within a single tile zoom. Also very noticeably less stable than a panTo animation over the same distance.

**Comment by diachedelic** [comment]
I am having this same isssue, I have [created a fiddle](https://plnkr.co/edit/RjI3J6PIdO36zVi1qyM3?p=preview) which displays the issue for me (MacOS Chrome 65.0.3325.181 (Official Build) (64-bit))

**Comment by perliedman** [comment]
@diachedelic thanks, that's a very illustrative example. I was able to pinpoint at least part of the issue from using your code.

[@mourner's comment](https://github.com/Leaflet/Leaflet/issues/3194#issuecomment-72756793) seems to be correct: just as an experiment, I removed some `round()` calls in the code (`GridLayer`'s `_setZoomTransform`, `Map`'s `_getNewPixelOrigin` and `Marker`'s `update` and `_animateZoom`) - this _almost_ resolves the issue.

However, I making these changes would likely break all sorts of stuff, see for example #6101, #3575 and #6069.

**Comment by perliedman** [comment]
@diachedelic yes, the marker is also vibrating. I think the main issue is that there are (at least) three different values being rounded: the map's pixel origin, the tile layer's transform and the marker's position/transform. Since these depend on each other, they interact in a complex way depending on if each one of the is rounded up or down, so that the result is a visible vibration.

**Comment by diachedelic** [comment]
It seems that when 'duration' is not specified, `flyTo` decides on a duration which suits the distance the map will move (zero distance to move means the animation does not occur, thereby avoiding this vibration).  In reality, I don't really want to set a static duration for `flyTo` - I just want to speed it up or slow it down by some factor.  Perhaps there could be an option for that?
